test/test.sh

```diff
@@ -1272,6 +1272,30 @@ test_no_database_auth_user() {
 	return 0
 }
 
+test_no_database_scram_auth_success() {
+	# Testing what happens on successful SCRAM auth connection to non-existent DB
+	# Segfaults have been seen after mock authentication was put in place
+
+	admin "set auth_type='md5'"
```

Comment:
After testing, it seems that `md5` was used here on purpose to trigger the bug. Using `scram-256` as auth_type doesn't crash. Would be good to add this detail to the comment above.

Reply:
I reran the tests using the current master branch of `pgbouncer` and saw the segfault occur when `auth_type` was set to scram or md5 with a scram PW. I've added more detail about the source of the bug to the tests though